crates/uv-resolver/src/pubgrub/package.rs

```diff
@@ -13,7 +13,6 @@ use crate::resolver::Urls;
 ///    package (e.g., `black[colorama]`), and mark it as a dependency of the real package (e.g.,
 ///    `black`). We then discard the virtual packages at the end of the resolution process.
 #[derive(Debug, Clone, Eq, Derivative)]
-#[derivative(PartialEq, Hash)]
```

Comment:
If we use the default ones, two requirements with different sources get flagged as different packages, and then you get duplicates in the output. Replacing them with hand-tooled options isn't the best of plans, but it is one that works and I don't have a better idea so far.